### PR TITLE
Update database example to include book creation

### PR DESCRIPTION
**Pull request by faun**
The database article was woefully incomplete and required an existing SQLite database with pre-existing data, so I created an endpoint for `POST /books` to create a book and moved the root route to `GET /books` for consistency. Also included are instructions for using `curl` to create and fetch the listing of books.

### CODE REVIEW

**Comment by faun** [comment]
I'm not sure if there are patterns for create JSON messages for endpoints, so I'm certainly open to code review on this. Also, the POST endpoint returns a 200, but it probably should return a 201.

**Comment by faun** [comment]
This code in this PR has been updated with the help of @ardan-bkennedy and @freeformz.

File: en/databases/README.md [hunk 1]
```diff
@@ -6,48 +6,155 @@ SQL databases. In this example we will connect to a SQLite database, but the
...
+
+		books = append(books, b)
 	}
 
-	return db
+	return books, nil
+}
+
+func createBook(db *sql.DB, b book) error {
+	q := fmt.Sprintf("INSERT INTO books(title, author) VALUES ('%s', '%s')", b.Title, b.Author)
```
Comment (freeformz) [inline en/databases/README.md]: This can actually lead you to an SQL injection. \o/

Comment (faun) [inline en/databases/README.md]: Good point.

Comment (freeformz) [inline en/databases/README.md]: Should probably be:

```go
q := "INSERT INTO books(title, author) VALUES ('?', '?')"
if _,err := db.Exec(q, b.Title, b.Author); err != nil {
...
```

Although I haven't tested it.

Comment (freeformz) [inline en/databases/README.md]:
```go
q := "INSERT INTO books(title, author) VALUES ('$1', '$2')"
if _,err := db.Exec(q, b.Title, b.Author); err != nil {
...
```

Comment (faun) [inline en/databases/README.md]: Fixed and squashed.

Comment (freeformz) [inline en/databases/README.md]: Don't see the changes. Did you push?

**Comment by faun** [comment]
It bothers me that the example file is entirely the same as the code that's in the readme. Perhaps I should break out the examples into manageable code pieces in the readme. Or perhaps even better, breaking the relevant concepts into distinct sub-sections unto themselves within the database section.

**Comment by freeformz** [comment]
+1

**Comment by faun** [comment]
Let's get this merged and then we can see about breaking this into smaller, more digestible pieces.

**Comment by freeformz** [comment]
:+1: /cc @carlisia

File: en/databases/README.md [hunk 2]
```diff
@@ -6,48 +6,154 @@ SQL databases. In this example we will connect to a SQLite database, but the
...
+
+		books = append(books, b)
 	}
 
-	return db
+	return books, nil
+}
+
+func createBook(db *sql.DB, b book) error {
+	q := "INSERT INTO books(title, author) VALUES ('$1', '$2')"
```
Comment (freeformz) [inline en/databases/README.md]: the single quotes around $1 and $2 need to be dropped.

Comment (faun) [inline en/databases/README.md]: Done.

Comment (faun) [inline en/databases/README.md]: I'm now regretting not adding tests for this example.

**Comment by freeformz** [comment]
We basically implemented 2 of the 3 exercises: https://github.com/gobridge/bwag/pull/31/files#diff-3db667871d986dcdd029fd299e8eab30R161

**Pull request merged**